### PR TITLE
FC-1187 Add Clojure api overview documentation

### PR DESCRIPTION
**Pull request by dpetran**

### CODE REVIEW

File: src/content/1.0.0/library/clojure/clojure-overview.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+[com.fluree/db "1.0.0-rc22"]
+```
+
+deps.edn -
+
+```all
+com.fluree/db {:mvn/version "1.0.0-rc22"}
+```
+
+Require the api and establish a connection to your Fluree server:
```
Comment (bplatz) [inline src/content/1.0.0/library/clojure/clojure-overview.md:35]: I think we should mention a previous step which is to ensure you are running a Fluree ledger.

i.e.:
Ensure you have a running Fluree Ledger service. Fluree can be downloaded <here>, unzip and run ./fluree-start.sh in the directory, or run from docker with the command <docker command here>. More detailed instructions can be found <here - assuming we have a page somewhere on running Fluree>.

File: src/content/1.0.0/library/clojure/clojure-overview.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+@(fdb/new-ledger conn ledger)
+```
+
+### Transacting
+
+`transact` is used to make any changes to the ledger, either adding data or schema. It's all data, and it all uses the same api to effect changes.
+
+#### Schema
+
+All data transacted into a ledger must conform to a schema. There are some built-in schemas that every ledger starts with that form very useful building blocks for constructing your own. See the [Schema](/docs/1.0.0/schema/overview) documentation for more details.
```
Comment (bplatz) [inline src/content/1.0.0/library/clojure/clojure-overview.md:55]: In the current version there is not built-in schemas. That is coming with the feature/prefix branch.

I'd suggest rewording this to something like:
All data transacted into a ledger must conform to a schema. We'll transact a basic schema below. See the [Schema] documentation for more details.

File: src/content/1.0.0/library/clojure/clojure-overview.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+#### Analytical queries
+
+The syntax for these queries is similar to datalog and eql. These don't support keywords for matching on flake values in the `:where` clause.
+
+The `query` function takes a `db` value to execute a query against, as well as a query map. See the docs for [Analytical queries](/docs/1.0.0/query/analytical-query) and [Basic queries](/docs/1.0.0/query/overview) for more details.
+
+```all
+;; all books
+@(fdb/query (db conn ledger) {:select {"?book" [:*, {:book/authors [:*]}]}
```
Comment (bplatz) [inline src/content/1.0.0/library/clojure/clojure-overview.md:158]: This would be (fdb/db conn ledger) - same as below.

File: src/content/1.0.0/library/clojure/clojure-overview.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+### Querying
+
+#### Analytical queries
+
+The syntax for these queries is similar to datalog and eql. These don't support keywords for matching on flake values in the `:where` clause.
+
+The `query` function takes a `db` value to execute a query against, as well as a query map. See the docs for [Analytical queries](/docs/1.0.0/query/analytical-query) and [Basic queries](/docs/1.0.0/query/overview) for more details.
+
+```all
+;; all books
```
Comment (bplatz) [inline src/content/1.0.0/library/clojure/clojure-overview.md:157]: I'd suggest perhaps adding above this:

;; store the current immutable database as a variable. See API docs for getting a prior version of a db, applying permissions to a db, and other options.

(def db (fdb/db conn ledger))

File: src/content/1.0.0/library/clojure/clojure-overview.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+```
+
+#### History queries
+
+[History queries](/docs/1.0.0/query/history-query) can show you the raw history of a subject. Use `prettyPrint` to get back maps of predicate names to object values instead of raw flake data.
+
+```all
+;; see the history of the "Cryptonomicon" book.
+@(fdb/history-query (db conn ledger) {:history ["book/title" "Cryptonomicon"] :prettyPrint true})
```
Comment (bplatz) [inline src/content/1.0.0/library/clojure/clojure-overview.md:174]: We could consider adding :pretty-print as the Clojure option and then in the Clojure API convert this to :prettyPrint. Same with other camel-case options. Not needed right now, but might be a good direction.

Comment (dpetran) [reply src/content/1.0.0/library/clojure/clojure-overview.md:174]: Looking into it, it looks like `block-query` and `history-query` both have special handling for kebab casing `:pretty-print` already, so I'll update the docstrings and the example.

As a side note, should we make pretty printing the default? maybe have a `:flakes` flag instead?

File: src/content/1.0.0/library/clojure/clojure-overview.md (hunk 4 above)
Comment (dpetran) [reply src/content/1.0.0/library/clojure/clojure-overview.md:157]: I got a little bitten by this as I was first learning the api - I had `def`ed the db value and I was bouncing back between transacting and querying and never seeing any of my updates until I realized that the `(db ,,,)` call gives you an immutable snapshot. I like adding this verbiage because if I'd read about it then I wouldn't have been surprised, but I think we should still use the full `(fdb/db conn ledger)` in the example queries instead of hardcoding the cached db value.

File: src/content/1.0.0/library/clojure/clojure-overview.md (hunk 5 above)
Comment (bplatz) [reply src/content/1.0.0/library/clojure/clojure-overview.md]: I can't recall everything pretty-print does. By the name I'd expect it to make things easier for a human but less so for a computer. I assume at least one thing is to fill in the full predicate names - but a computer will just turn around and need the predicate IDs - but I think I'd have to look at this a bit more and perhaps talk about the right behavior and ensure it is consistent across different uses.

File: src/content/1.0.0/library/clojure/clojure-overview.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+#### Other query languages
+
+In addition to the FlureeQL syntax used above, Fluree also supports queries in other query languages: [SQL](/docs/1.0.0/query/sql), [SPARQL](/docs/1.0.0/query/sparql), [GraphQL](/docs/1.0.0/query/graphql). These can be used from Clojure by supplying a database value and a query string, or, in the case of GraphQL, a query map.
+
+```all
+@(fdb/sql (fdb/db conn ledger) "select * from book")
+
+@(fdb/sparql (fdb/db conn ledger) "SELECT ?title WHERE { ?book fd:book/title ?title; fd:book/tags \"sci-fi\".}")
```
Comment (Tr3yb0) [inline src/content/1.0.0/library/clojure/clojure-overview.md:196]: A very tiny thing, but you tag you used above was sci-foo 😄

**Review by Tr3yb0 — APPROVED** [review]

**Comment by dpetran** [comment]
Are there any more blockers on this PR?

**Comment by Tr3yb0** [comment]
nope i thought when i merged it, it closed the PR.. the changes have been pushed to the public deploy

**Comment by dpetran** [comment]
I talked to @cap10morgan about that, I'm closing the pom.xml PR (#84) and we'll explore using tools.build to generate the correct pom sometime in the future.

**Comment by Tr3yb0** [comment]
Ok, I am going to remove the badge from the page then. We can add it back when we get everything set up
